code/frameworks/angular/src/client/docs/compodoc.ts

```diff
@@ -20,6 +20,9 @@ export const isMethod = (methodOrProp: Method | Property): methodOrProp is Metho
   return (methodOrProp as Method).args !== undefined;
 };
 
+export const isRequired = (prop: Property): object =>
+  prop.hasOwnProperty('required') && prop.required ? { required: true } : {};
+
```

Comment:
This seems a little misleading, with how it is named. Typically a is* represents a boolean. would it be better to change this to returning a boolean and changing the places this is used to `required: isRequired(property)`. It doesn't change the end result, but I wonder if it would make the code more clear and the method more reusable. I don't have a strong opinion either way, it just stuck out as odd when I was reading the code and initially thought it was applying the spread operator to a boolean.

Reply:
Yeah good shout, I'll rework it a bit 👍

Reply:
One thing I want to be careful of though, re: returning a boolean, was most of the time it will return false (most `@Inputs()` don't have the `required: true` prop. Which is why i used the spread operator checking if the prop exists AND if it was set to true, and if not, return an empty object essentially not adding anything onto the argType. 

It felt a bit redundant enhancing the argTypes with lots of `required: false` so prob want to avoid that if possible

Reply:
That makes sense. When I want to only add a property, I normally do something like `...(isRequired(property) ? { required: true } : {}). It is fairly easy to read, but I know it doesn't look great. If we want to use the method, like you had, a rename or additional method may work. This isn't a major piece of code that would be used outside of this file, so it doesn't matter much. An `isRequired` method that returns a boolean may not be needed, unless we need it for that other `required` property.

Reply:
I've updated that now, subtle difference but a bit clearer 😄